bluesky_adaptive/adjudicators/base.py

```diff
@@ -0,0 +1,236 @@
+        r = re_manager.item_add(plan, pos="back")
+        logger.debug(f"Sent http-server request by adjudicator\n." f"Received reponse: {r}")
+
+    def server_registrations(self) -> None:
+        """
+        Method to generate all server registrations during agent initialization.
+        This method can be used in subclasses, to override or extend the default registrations.
+        """
+        self._register_method("make_judgements", "_make_judgments_and_add_to_queue")
+        self._register_property("prompt_judgement")
+
+    def _make_judgments_and_add_to_queue(self):
+        """Internal wrapper for making judgements, validating, and adding to queue."""
+        judgments = self.make_judgments()
+        for judgment in judgments:
+            if not isinstance(judgment, Judgment):
+                judgment = Judgment(*judgment)  # Validate
+            self._add_suggestion_to_queue(**judgment.dict())
+
+    @abstractmethod()
+    def make_judgments(self) -> Sequence[Tuple[API_Threads_Mixin, str, Suggestion]]:
+        """Instance method to make judgements based on current suggestions.
+        The returned tuples will be deconstructed to add suggestions to the queue.
+        """
+        ...
+
+
+class AgentByNameAdjudicator(AdjudicatorBase):
+    """Adjudicator that only allows messages from a set primary agent, and uses a single qserver.
+    Parameters
+    ----------
+    qservers :  dict[str, API_Threads_Mixin]
+        Dictionary of objects to manage communication with Queue Server. These should be keyed by the beamline TLA
+        expected in AdjudicatorMsg.suggestions dictionary.
+    """
+
+    def __init__(self, *args, qservers: dict[str, API_Threads_Mixin], **kwargs):
+        self._primary_agent = ""
+        self._re_managers = qservers
+        super().__init__(*args, **kwargs)
+
+    @property
+    def primary_agent(self):
+        return self._primary_agent
+
+    @primary_agent.setter
+    def primary_agent(self, name: str):
+        self._primary_agent = "str"
+
+    def server_registrations(self) -> None:
+        self._register_property("priamry_agent")
+        super().server_registrations()
+
+    def make_judgments(self) -> Sequence[Tuple[API_Threads_Mixin, str, Suggestion]]:
+        judgments = []
+
+        if self.primary_agent not in self.agent_names:
+            logger.debug(f"Agent {self.primary_agent} not known to the Adjudicator")
+        else:
+            adjudicator_msg = self.current_suggestions[self.primary_agent]
+            for key, manager in self._re_managers.items():
+                suggestions = adjudicator_msg.suggestions.get(key, [])
+                for suggestion in suggestions:
+                    judgments.append(
+                        Judgment(re_manager=manager, agent_name=self.primary_agent, suggestion=suggestion)
+                    )
+        return judgments
+
+
+class NonredundantAdjudicator(AdjudicatorBase):
```

Comment:
I suspect we will want a "near by" version of this too where instead of a hash function it is a "give me a location in ND space".

Reply:
My intent was to leave this up to the beamline. 
So they could hash a plan as `f"{tla} {suggestion.plan_name} {np.floor(suggestion.plan_args[0] /self.resolution)}"`

Reply:
Follow up in #18